src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+                    case DAY_OF_MONTH:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.DAY_OF_MONTH_RANGE;
+                    case DAY_OF_YEAR:
+                        return special ?
+                                    ValueRange.of(getDayOfYear(), getDayOfYear()) :
+                                    isLeapYear() ?
+                                            InternationalFixedChronology.DAY_OF_YEAR_LEAP_RANGE :
+                                            InternationalFixedChronology.DAY_OF_YEAR_NORMAL_RANGE;
+                    case EPOCH_DAY:
+                        return InternationalFixedChronology.EPOCH_DAY_RANGE;
+                    case ERA:
+                        return InternationalFixedChronology.ERA_RANGE;
+                    case MONTH_OF_YEAR:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.MONTH_OF_YEAR_RANGE;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        return super.range(field);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    public int getAlignedDayOfWeekInMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfMonth() - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedDayOfWeekInYear() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfYearAdjusted() - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedWeekOfMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfMonth() - 1) / lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedWeekOfYear() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfYear() - 1) / lengthOfWeek()) + 1;
+    }
+
+    /**
+     * Returns the day of the week represented by this date.
+     * <p/>
+     * This returns the ordinal of the enum {@link DayOfWeek}.
+     * Day-of-week do not match those of the ISO calendar system.
+     * In particular, each months starts with Sunday.
+     * Leap-day and year-day are not considered week-days, thus return 0.
+     *
+     * @return the day of the week enumeration: between 1 (Monday) and 7 (Sunday), or 0 (leap-day, year-day)
+     */
+    @Override
+    public int getDayOfWeek() {
+        if (isLeapDay() || isYearDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted();
+
+        return 1 + ((5 + doy) % 7);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    long getProlepticMonth() {
```

Comment:
You should be returning the values returned in `getCalculatedProlepticMonth()` here instead (ie, that leap-day is accounted with the 7th month, and year-day with 13th).  Ditch the other method.